src/visitor.rs

```diff
@@ -592,6 +606,10 @@ impl<'b, 'a: 'b> FmtVisitor<'a> {
             ctx.snippet_provider,
             ctx.report.clone(),
         );
+        visitor
+            .skip_macro_names
+            .borrow_mut()
+            .append(&mut ctx.skip_macro_names.borrow_mut());
```

Comment:
I think that this will leave `ctx.skip_macro_names` empty ([`append`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.append)).

Reply:
Actually I intentionally didn't use clone. Because ctx.skip_macro_names is not used after appending to skip_macro_names.
But on second thought, it's not a good idea. I thinak this logic is confusing and will cause bug in near future.
So I fixed it. Thank you.

Reply:
Thank you for providing us with the context.

> ctx.skip_macro_names is not used after appending to skip_macro_names.

I think that there are circumstances where `ctx.skip_macro_names` gets reused :)
One such case is when we create multiple visitors from the same context:

```rust
let pair = (
  || {
    foo!();
    bar!();
  },
  || {
    foobar!();
    skip_this! ( );
  },
}
```

IIUC, in the above example, two `FmtVisitor` will be created from a single `RewriteContext` (via `format_expr` on the right hand side of `let`).

Reply:
I tested it now, and it was failed. Your opinion is right. Thanks a lot.